### ISSUE TITLE
Notebook Weekly Meetings

### ISSUE THREAD

**Issue by jtpio** [jupyterlab/team-compass#133]
Here is a record of the [agenda/minutes](https://hackmd.io/gl6x_SL0S_aYVn0Y0lSBpQ?both) for the Notebook weekly meeting, happening 30 minutes before the [weekly JupyterLab meeting](https://github.com/jupyterlab/jupyterlab#weekly-dev-meeting). 

- When: Wednesdays [8:30 am Pacific Time](https://dateful.com/convert/san-francisco-california?t=830am)
- Where: [`jovyan` Zoom](https://zoom.us/my/jovyan?pwd=c0JZTHlNdS9Sek9vdzR3aTJ4SzFTQT09)
- What: [Meeting notes](https://hackmd.io/gl6x_SL0S_aYVn0Y0lSBpQ?both) 

During the last meeting, we discussed posting the notes on a new issue in the team compass. Here is the record of these notes.

Feel free to comment here if you think the issue should be open somewhere else.

## Meeting Notes

- [26 January 2022](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-1022489712)
- [19 January 2022](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-1016703643)
- [12 January 2022](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-1011295405)
- [05 January 2022](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-1006358769)
- [15 December 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-995593150)
- [08 December 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-989018555)
- [01 December 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-983919646)
- [24 November 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-982685099)
- [17 November 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-972888376)
- [10 November 2021](https://github.com/jupyterlab/team-compass/issues/133#issuecomment-972888146)

**Comment by jtpio** [comment]
## November 10, 2021

### attendees

| name | org | username |
|------|-----|----------|
|tony fast | quansight |@tonyfast|
| Jason Weill | AWS | @jweill-aws |
| Zach Sailer | Apple | @Zsailer |
| Jeremy Tuloup | QuantStack | @jtpio |
| Rick McGeer | engageLively | @rickmcgeer |
| Frederic Collonval | QuantStack | @fcollonval |
| Peter Wang | Anaconda | @pzwang |
| Kevin Goldsmith | Anaconda | @kevingoldsmith |
| Jessica Xu | Quansight | @jess-x|



### agenda

* define new goals for the meeting
* discuss the merge strategy for legacy classic and retro
    * where to handle discussions knowing we're replacing one technology with another.
    * Technical steps/questions
        * Do we move retrolab into jupyter/notebook?
        * Dis-entangling NBClassic, JupyterLab, and Notebook
            * NBClassic depends on Notebook and JupyterLab depends on NBClassic.
            * Making RetroLab become Notebook 7.x forces JupyterLab to drop NBClassic as dependency *or* NBClassic drops it's notebook dependency.
            * Pin nbclassic to 6.x.
* create issues and organize the project board https://github.com/orgs/jupyterlab/projects/2
* status updates - goals, blockers, availability

**Comment by jtpio** [comment]
## November 17, 2021

### attendees

| name | org | username |
|------|-----|----------|
|tony fast | quansight |@tonyfast|
| A. T. Darian | Two Sigma | @afshin |
| Peter wang | Anaconda | @pzwang | 
| Jeremy Tuloup | QuantStack | @jtpio |
| Sylvain Corlay | QuantStack | @SylvainCorlay |
| Frederic Collonval | QuantStack | @fcollonval |
| Kevin Goldsmith | Anaconda | @KevinGoldsmith |

### agenda

* where do meeting notes go?
    * add an issue to the team-compass "notebook weekly minutes"
* issue tag for sunrise related tasks
    * reusing `tag:Feature Parity`
* transfer organizations
    * jupyter/notebook is the repo for the notebook package
    * https://github.com/jupyterlab/team-compass/issues/132
* update on pre release work
    * blocked on dependencies
* update on communications work (eg issues, issue templates)
* Notebook 7 JEP draft updates, governance updates

### new pull requests and issues

- Enable scrolling outputs by default: https://github.com/jupyterlab/retrolab/pull/288
    - Something that should go in JupyterLab (feature parity with classic)
    - Might be a blocker for adoption for Data 8

**Comment by jtpio** [comment]
## November 24, 2021

### attendees

| name | org | username |
|------|-----|----------|
|tony fast | quansight |@tonyfast|
| Rick McGeer | engageLively | @rickmcgeer |
|Isabela Presedo-Floyd | Quansight Labs | @isabela-pf |
| A. T. Darian | Two Sigma | @afshin |
| Frederic Collonval | QuantStack | @fcollonval |
| Sylvain Corlay | QuantStack | @SylvainCorlay |
| Kevin Goldsmith | Anaconda | @KevinGoldsmith |


### agenda
 
* [[DRAFT] JEP for Notebook v7](https://github.com/jupyter/enhancement-proposals/pull/79)


- Frederic: JupyterLab 3.x or 4.0 for notebook 7 [issue](https://github.com/jupyterlab/jupyterlab/issues/11464)   
  Currently RetroLab (to-be notebook 7) is based on lab 3.x branch. But some new features will be needed for it:   
  - Customize widget position to fit with notebook layout
  - Add cells toolbar ([TBC] bringing [jlab-enhanced-cell-toolbar](https://github.com/jupyterlab-contrib/jlab-enhanced-cell-toolbar) in Lab)
  - Add tour (bringing [jupyterlab-tour](https://github.com/jupyterlab-contrib/jupyterlab-tour) in Lab)   
  **Cons for 3.x**
  - If we do a release 3.x for it, this will strip more reasons to update to JLab 4.
  - If it is linked to 4, this avoid releasing notebook 8 after JLab 4 :wink:
  - An excuse for extensions authors to modify their extension to be compatible because the API has changed.   
  **Pros for 3.x**  
  - Less breaking changes?  
    
  [Afshin] it makes sense regarding the amount of work to be done for notebook v7 that it will be based on JLab 4. Moreover it will be better to not _annoy_ extension authors that just got converted to NPM world to modify again their extensions because we bumped the major version.

**Comment by jtpio** [comment]
## December 1, 2021

### attendees

| name | org | username |
|------|-----|----------|
|tony fast | quansight |@tonyfast|
| Zach Sailer | Apple | @Zsailer | 
| R Ely | Bloomberg | @ohrely |
| Gabriel Fouasnon | Quansight | @gabalafou |
| Theodore Aptekarev| No org 🤷‍♂️ | @piiq |
| Jeremy Tuloup | QuantStack | @jtpio |
|Isabela Presedo-Floyd| Quansight Labs | @isabela-pf |
|Fabio Pliger| Anaconda | @fpliger |

### agenda
 
* Prepare nbclassic for Notebook 7.
    1. nbclassic the classic notebook ui running on jupyter server 
        * Release Notebook static/templates under a separate package (possible on NPM).
    2. nbclassic automatically shims config
        * Copy traits from jupyter/notebook's NotebookApp into nbclassic package
    3. no nbclassic
        * No changes needed
* Drop [old notebook traitlets shim](https://github.com/jupyterlab/jupyterlab/blob/7a2878ab3556e9bde7e68b487bbc3c622c6fe604/jupyterlab/labapp.py#L478) from JupyterLab?

**Comment by Zsailer** [comment]
## December 8th, 2021

### attendees

| name | org | username |
|------|-----|----------|
| Zach Sailer | Apple | @Zsailer | 
| Jason Weill | AWS| @jweill-aws |
| Kevin Goldsmith | Anaconda | @KevinGoldsmith |
| Loic Huder | ESRF | @loichuder |
| Fabio Pliger | Anaconda | @fpliger |
| Brian Granger | AWS | @ellisonbg
| Jeremy Tuloup | QuantStack | @jtpio | 
| Piyush Jain | AWS | @3coins |

There were ~5 others in the meeting that we're captured in this list. 

### agenda
 
* From last week
    * Prepare nbclassic for Notebook 7.
        * Discussion opened here: https://github.com/jupyterlab/nbclassic/issues/81
    * Drop [old notebook traitlets shim](https://github.com/jupyterlab/jupyterlab/blob/7a2878ab3556e9bde7e68b487bbc3c622c6fe604/jupyterlab/labapp.py#L478) from JupyterLab?
        * https://github.com/jupyterlab/jupyterlab/issues/11612

* Jupyter Notebook [v7 Enhancement Proposal](https://github.com/jupyter/enhancement-proposals/pull/79)
    * Entered voting phase this past week
    * [Open discussion](https://github.com/jupyter/enhancement-proposals/pull/79#discussion_r763481751) on the guiding principles for Notebook v7
        * Some voiced the following concern
            * As currently written, there appears to be a tilt towards favoring a "future" experience over preserving the current Notebook experience.
        * A proposed alternative: "Any user facing difference in behavior from V6 is bug".
            * Are we willing to make this strong of a statement?
                * There was concern raised about the strength of this claim.
            * Looking 5 years out, a vast majority of our users are not using Jupyter today. 
                * Who do we optimize for—today's users or tomorrow's users?
        * Keep in mind, the spirit of the JEP is captured by the last principle. 
            * "Have we done everything possible to minimize the experience gap for users of Notebook v6 (while recognizing, given other constraints, that we can't always eliminate it completely)?"
            * In specific cases mentioned—i.e. accessibility, maintainability, and security—we **cannot** optimize for user experience gap between v6 and v7.
    * Notes about voting and discussion.
        * JEP is in final round of comments.
        * Complete concensus is not a requirement for JEP approval. 
        * The Steering Council's vote determines the outcome.
        * Final comments should happen before December 13th.

**Comment by jtpio** [comment]
## December 15th, 2021

### attendees

| name | org | username |
|------|-----|----------|
| tonyfast | quansight | @tonyfast | 
| Zach Sailer | Apple | @Zsailer |
| Jeremy Tuloup | QuantStack | @jtpio |
| David Brochart | QuantStack | @davidbrochart |
| Fabio Pliger  | Anaconda   | @fpliger | 
| A. T. Darian | Two Sigma | @afshin |
| Michal Krassowski | Oxford | @krassowski |
| Craig Citro | Google | @craigcitro |
| Gabriel Fouasnon | Quansight | @gabalafou |

### agenda
 
* [Accessibility hands on workshop tomorrow @ 9am PST](https://hackmd.io/@p5jde6ivTRa6LnqFD8l8wQ/Hk42FoH_Y)
* [name=Zach]
    * Current status of the JEP:
        * 10 yes, 1 no, 3 abstain, 4 absent.
        * What is needed to pass? 
    * Need some clarity from last week's discussion:
        * Question from last week: Can Notebook v6 be installed in the same environment as Notebook v7?
            * Polled some people since last week, and the answers were mixed.
        * If not, I think we should provide clear messaging in Notebook v7 (client-side and in terminal)
            * how to downgrade back to Notebook v6 if absolutely needed
            * where to report changes that break workflows.
        * Upgrading to Notebook v7 is not _really_ an opt-in action. 
            * Most people call `pip install notebook` without an explicit version pin.
            * Defaults to Notebook v7 (not necessarily a problem) 
            * Users have to opt-out to get the Notebook v6 experience.
            * Darian's opinion: I think we should allow *security* patches to notebook v6 but no other releases that add features or even really bug fixes unless they are truly trivial. I think `nbclassic` should be deprecated. This is my opinion but it did seem to be shared at the last governance call where I brought up the subject. I just don't want to attach other names to ideas I wrote down off the cuff.
        * When do we cut a pre-release?
            * As soon as the JEP is merged.
            * This is low risk, because `pip install --pre` is an explicit, opt-in action.

**Comment by jtpio** [comment]
## January 5th, 2022

### attendees

| name | org | username |
|------|-----|----------|
| tonyfast | quansight | @tonyfast | 
| Jason Weill | AWS | @jweill-aws | 
| Zach Sailer    | Apple  | @Zsailer | 
| Isabela Presedo-Floyd | Quansight Labs | @isabela-pf | 
| Jeremy Tuloup | QuantStack | @jtpio | 
| David Brochart | QuantStack | @davidbrochart |
| Kevin Bates | IBM | @kevin-bates |
| Frederic Collonval | QuantStack | @fcollonval |

### agenda
 
* JEP accepted!
* Remove Bower dependency PR in jupyter/notebook (https://github.com/jupyter/notebook/pull/6213)
* nbviewer.org home page still has many broken links ([nbviewer#826](https://github.com/jupyter/nbviewer/issues/826))
* pre release
    * next version of nbclassic
    * retrolab
    * what is the dependency problem with nbclassic and retrolab?
        * https://github.com/jupyterlab/nbclassic/issues/81
    * how to pre-release?
        there are 2 goals:
        1. get a pre release sooner than later
        2. cut a proper pre release in a few months

        we'll start a discussion around zachs proposed solution, and go from there.
        
        * from notebook
            * add retrolab as a dependency to notebook using entry points

**Comment by jtpio** [comment]
## January 12th, 2022

### attendees

| name | org | username |
|------|-----|----------|
| tonyfast | quansight | @tonyfast | 
| Jason Weill | AWS | @jweill-aws|
| Sylvain Corlay | QuantStack | @SylvainCorlay |
| A. T. Darian | Two Sigma | @afshin |
| Gabriel Fouasnon | Quansight | @gabalafou |
| Jeremy Tuloup | QuantStack | @jtpio |
| Carlos Herrero | QuantStack | @hbcarlos |
| Craig Citro | Google | @craigcitro |

### agenda

* pre release issue https://github.com/jupyter/notebook/issues/6266
    * endpoints need to be shuffled
    * rename retrolab to notebook-7 (temporarily)?
    * put retrolab and notebook 6.x in separate branches of jupyter/notebook repo.
* Notebook maintenance:
    * Releasing 6.4.7 as we speak
    * CSS regression from after switching to sanitize-html
    * Thanks to Mike for helping debug!
* Software Steering Council + decision-making body for Notebook Subproject
* bower replacing npm pr still open
    * good thing for anaconda to work on?

**Comment by jtpio** [comment]
## January 19th, 2022

### attendees

| name | org | username |
|------|-----|----------|
| tonyfast | quansight | @tonyfast | 
| A. T. Darian | Two Sigma | @afshin |
|Zach Sailer | Apple | @Zsailer |
| Jason Weill | AWS | @jweill-aws |
| Jeremy Tuloup | QuantStack | @jtpio |
| Kevin Goldsmith | Anaconda | @kevingoldsmith |
| Damián | 2i2c & MADS | @damianavila |

### agenda

* decision-making body update
* pre-release
    * Created a 6.4.x branch in jupyter/notebook
    * `master` branch becomes retrolab via a PR
            * Should we rename `master` to `main`? :ballot_box_with_check: 
    * Some work started in the RetroLab repo to get the CI checks to pass: https://github.com/jupyterlab/retrolab/pull/323
    * Blocker? Drop `nbclassic` in JupyterLab: https://github.com/jupyterlab/jupyterlab/issues/11612
        * Or implement option 2 mentioned in https://github.com/jupyterlab/nbclassic/issues/81#issue-1070846121 to move the shims to a separate package
* We need a `team-compass` repo. What do you think of `github/jupyter/notebook-team-compass` as the URL?
* Need help with Notebook 6.
    * Issue Triaging; let's label all issue with "Notebook 6"?
    * Dropping bower; Anaconda team can possibly pick this up once they get a team up and running.
    * Community workshop for issue triaging

**Comment by jtpio** [comment]
## January 26th, 2022

### attendees

| name | org | username |
|------|-----|----------|
| tonyfast | quansight | @tonyfast | 
| Kevin Goldsmith | Anaconda | @kevingoldsmith |
| Zach Sailer | Apple | @Zsailer |
| A. T. Darian | Two Sigma | @afshin |
| Jason Weill| AWS | @jweill-aws |
| Fabio Pliger | Anaconda | @fpliger |
| Kevin Bates | IBM | @kevin-bates |
| Jeremy Tuloup | QuantStack | @jtpio |
| Nick Bollweg | Georgia Tech | @bollwyvl |

### agenda

* [Jupyter Notebook Team Compass Page](https://jupyter-notebook-team-compass.readthedocs.io/en/latest/#) is live!
    * https://github.com/jupyter/notebook-team-compass/
* Created `notebook_shim` this last week: https://github.com/jupyterlab/notebook_shim
    * Nbclassic currently does two major things
        1. Shim config and traits from notebook to jupyter_server
        2. Adds notebook handlers
    * This new package handles (1).
    * Allows JupyterLab to drop its indirect dependency on the `notebook` package.
* Decision-making body update -- relevant to team-compass update
    * team compass will need to be update with members
    * onboarding first team members: https://github.com/jupyter/notebook-team-compass/pull/4
* JupyterLab/Notebook triage meeting
    * https://hackmd.io/HaQ3S_nPSbqaRtk9rj59iA
    * references
        * https://www.kubernetes.dev/docs/guide/issue-triage/
        * https://github.com/moby/moby/blob/master/project/ISSUE-TRIAGE.md
        * [JupyterLab triage guidelines](https://jupyterlab.readthedocs.io/en/latest/developer/contributing.html#submitting-a-pull-request-contribution)

* MathJax3: https://github.com/jupyter/nbconvert/issues/1706

**Comment by jtpio** [comment]
@meeseeksdev please migrate to https://github.com/jupyter/notebook-team-compass

**Comment by Zsailer** [comment]
@meeseeksdev please migrate to jupyter/notebook-team-compass
